automation/taskcluster/release/convert_group_and_artifact_ids.py

```diff
@@ -0,0 +1,264 @@
+
+
+def extract_artifact_id(file_name):
+    return file_name.split('/')[-3]
+
+
+def process_pom_and_checksums(zip_in, zip_out, zip_info, new_file_name, conversion_dict):
+    raw_xml = zip_in.read(zip_info.filename)
+    pom = translate_pom_xml(raw_xml, conversion_dict)
+    zip_out.writestr(new_file_name, pom, compress_type=zipfile.ZIP_DEFLATED)
+
+    for hash_alg in SUPPORTED_HASHES:
+        sum = get_checksum(pom, hash_alg)
+        zip_out.writestr('{}.{}'.format(new_file_name, hash_alg), sum, compress_type=zipfile.ZIP_DEFLATED)
+
+
+def copy_file_into_zip(zip_in, zip_out, zip_info, new_file_name):
+    with tempfile.NamedTemporaryFile() as temp:
+        temp.write(zip_in.read(zip_info.filename))
+        temp.seek(0)
+        zip_out.write(
+            temp.name, arcname=new_file_name,
+            # TODO: Figure out why zip_info.compress_type is invalid
+            compress_type=zipfile.ZIP_DEFLATED
+        )
+
+
+def get_checksum(data, hash_alg):
+    h = hashlib.new(hash_alg)
+    h.update(data)
+    return h.hexdigest()
+
+
+def translate_pom_xml(raw_xml, conversion_dict):
+    root = ElementTree.fromstring(raw_xml)
+    convert_xml_element(root, conversion_dict)
+
+    dependencies = root.find('m:dependencies', XML_NAMESPACES)
+    if dependencies is not None:    # Some packages may not have any dependencies
+        for dependency in dependencies.findall('m:dependency', XML_NAMESPACES):
+            convert_xml_element(dependency, conversion_dict)
+
+    new_xml = ElementTree.tostring(root, encoding='UTF-8')
+    log.debug('pom.xml modified. From: {}\n\nTo: {}'.format(raw_xml, new_xml))
+
+    return new_xml
+
+
+def convert_xml_element(element, conversion_dict):
+    group_id_element = element.find('m:groupId', XML_NAMESPACES)
+    old_group_id = group_id_element.text
+    artifact_id_element = element.find('m:artifactId', XML_NAMESPACES)
+    old_artifact_id = artifact_id_element.text
+
+    try:
+        new = conversion_dict[get_conversion_key(old_group_id, old_artifact_id)]
+        group_id_element.text = new['new_group_id']
+        artifact_id_element.text = new['new_artifact_id']
+    except KeyError:
+        pass
+
+
+async def async_main(session, task_id):
+    asyncQueue = taskcluster.aio.Queue(session=session)
+    task_definition = await asyncQueue.task(task_id)
+    upstream_zip_definitions = task_definition['payload']['upstreamZip']
+
+    upstream_zip_definitions = [
+        {
+            **definition,
+            'url': craft_taskcluster_download_url(definition['taskId'], definition['path']),
+            'abs_filename': craft_abs_filename(definition['path']),
+        }
+        for definition in upstream_zip_definitions
+    ]
+
+    await download_all_zip_artifacts(upstream_zip_definitions, session)
+
+    conversion_dict = build_conversion_dict(upstream_zip_definitions)
+
+    log.debug('Files will be converted based on this dictionary: {}'.format(conversion_dict))
+
```

Comment:
Nit: This blank line can go away since the log line describes the command below.